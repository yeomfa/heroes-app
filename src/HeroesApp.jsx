import { AuthProvider } from './auth';
import { AppRouter } from './router';

export const HeroesApp = () => {
  return (
    <AuthProvider>
      <AppRouter />
    </AuthProvider>
  )
}
