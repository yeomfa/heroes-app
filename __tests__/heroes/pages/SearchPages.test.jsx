import { fireEvent, render, screen } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import { SearchPage } from '../../../src/heroes';

const mockUseNavigate = jest.fn();

jest.mock('react-router-dom', () => ({
	...jest.requireActual('react-router-dom'),
	useNavigate: () => mockUseNavigate,
}))

beforeEach(() => {
	jest.clearAllMocks();
})

describe('Tests on <SearchPage />', () => {

	test('debe de mostrarse correctamente con valores por defecto', () => {

		const { container } = render(
			<MemoryRouter>
				<SearchPage />
			</MemoryRouter>
		);

		expect( container ).toMatchSnapshot();

	})

	test('debe de mostrar al batman y el input con el valor del queryString', () => {

		render(
			<MemoryRouter initialEntries={['/?q=batman']}>
				<SearchPage />
			</MemoryRouter>
		);

		const input = screen.getByRole( 'textbox' );
		expect( input.value ).toBe( 'batman' );

		const img = screen.getByRole( 'img' );
		expect( img.src ).toContain('/assets/heroes/dc-batman.jpg');

		const alertSearch = screen.getByLabelText('alert-search');
		expect( alertSearch.style.display ).toBe( 'none' );
		
	})

	test('debe de mostrar un error si no encuentr el hero (batman123)', () => {

		render(
			<MemoryRouter initialEntries={['/?q=batman123']}>
				<SearchPage />
			</MemoryRouter>
		);

		const alertError = screen.getByLabelText('alert-error');
		expect( alertError.style.display ).toBe( '' );
		
	})

	test('debe de llamar el navigate a la pantalla nueva', () => {

		render(
			<MemoryRouter>
				<SearchPage />
			</MemoryRouter>
		);

		const input = screen.getByRole('textbox');
		const form = screen.getByLabelText('form');

		fireEvent.change( input, { target: { value: 'goku' } } );
		fireEvent.submit( form );

		expect( mockUseNavigate ).toHaveBeenCalledWith('?q=goku');
		
	})

});
