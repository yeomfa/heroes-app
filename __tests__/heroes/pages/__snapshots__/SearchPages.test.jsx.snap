// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests on <SearchPage /> debe de mostrarse correctamente con valores por defecto 1`] = `
<div>
  <h1>
    Search
  </h1>
  <div
    class="col-5"
  >
    <form
      aria-label="form"
      class="d-flex"
    >
      <input
        autocomplete="off"
        class="form-control"
        name="searchText"
        placeholder="Search a hero"
        type="text"
        value=""
      />
      <button
        class="btn btn-primary"
        type="submit"
      >
        Search
      </button>
    </form>
  </div>
  <hr />
  <h4>
    Results
  </h4>
  <div
    class="row row-cols-1 row-cols-md-3 g-3"
  >
    <div
      aria-label="alert-search"
      class="alert alert-primary"
    >
      Search a hero
    </div>
    <div
      aria-label="alert-error"
      class="alert alert-danger"
      style="display: none;"
    >
      No hero with 
      <b />
    </div>
  </div>
</div>
`;
